weather/src/main/java/bot/WeatherApiClient.java

```diff
@@ -0,0 +1,74 @@
+package bot;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.util.zip.GZIPInputStream;
+
+public class WeatherApiClient implements WeatherClient {
+
+    private static final String HOST_URL = "http://api.weatherstack.com/current?access_key=";
+
+    private static final Logger LOGGER = Logger.getLogger(WeatherApiClient.class);
+    private static final OkHttpClient CLIENT = new OkHttpClient();
+
+    private String accessKey;
+
+    public WeatherApiClient(String accessKey) {
+        this.accessKey = accessKey;
+    }
+
+    @Override
+    public Weather byCity(String cityName) {
+        Weather currentWeather = null;
+
+        Request request = new Request.Builder()
+                .url(HOST_URL + accessKey + "&query=" + cityName)
+                .addHeader("Accept-Encoding", "gzip, deflate")
+                .build();
+
+        LOGGER.debug("Request: " + request.url());
+
+        try {
+            Response response = CLIENT.newCall(request).execute();
```

Comment:
```suggestion
        try (Response response = CLIENT.newCall(request).execute()) {
```
Responses must be closed.